Source/debug.cpp

```diff
@@ -294,6 +295,82 @@ std::string DebugCmdLoadMap(const string_view parameter)
 	return "Welcome to this unique place.";
 }
 
+std::string ExportDun(const string_view parameter)
+{
+	std::ofstream dunFile;
+
+	std::string levelName = fmt::format("{}-{}.dun", currlevel, glSeedTbl[currlevel]);
+
+	dunFile.open(levelName, std::ios::out | std::ios::app | std::ios::binary);
+
+	char padding = 0;
+	uint16_t width = DMAXX;
+	uint16_t height = DMAXY;
+	dunFile.write((char *)&width, 2);

```

Comment:
This is an unaligned read (UB) and not endian-correct.

Perhaps something like this:

```c++
void WriteLE16(std::fstream &out, uint16_t val)
{
	const uint16_t littleEndian = SDL_SwapLE16(val);
	char data[2];
	memcpy(data, &littleEndian, 2);
	out.write(data, 2);    
}
```

Reply:
It wasn't all that important since it's only for debugging, but since you provided a function I'll add it when I get home 😊

Reply:
There we go, much nicer :)